dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java

```diff
@@ -36,57 +36,31 @@
                                                                                                                                                         "service.ranking:Integer=9999" })
 public class CDIManagedObjectService implements ManagedObjectService {
 
-    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
-    private final AtomicServiceReference<DefaultManagedObjectService> defaultMOSRef = new AtomicServiceReference<DefaultManagedObjectService>("defaultManagedObjectService");
+    private DefaultManagedObjectService defaultMOS;
+    private CDIService cdiService;
 
     private CDIRuntime cdiRuntime;
 
     public void activate(ComponentContext cc) {
-        cdiServiceRef.activate(cc);
-        defaultMOSRef.activate(cc);
     }
 
     public void deactivate(ComponentContext cc) {
-        cdiServiceRef.deactivate(cc);
-        defaultMOSRef.deactivate(cc);
         cdiRuntime = null;
     }
 
-    @Reference(name = "defaultManagedObjectService", service = DefaultManagedObjectService.class, policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MANDATORY)
-    protected void setDefaultManagedObjectService(ServiceReference<DefaultManagedObjectService> ref) {
-        defaultMOSRef.setReference(ref);
+    @Reference(name = "cdiService", service = CDIService.class, cardinality = ReferenceCardinality.MANDATORY)
+    public void setCDIService(CDIService s) {
+        cdiService = s;
     }
 
-    protected void unsetDefaultManagedObjectService(ServiceReference<DefaultManagedObjectService> ref) {
-        defaultMOSRef.unsetReference(ref);
-    }
-
-    private DefaultManagedObjectService getDefaultManagedObjectService() {
-        return defaultMOSRef.getServiceWithException();
-    }
-
-    @Reference(name = "cdiService", service = CDIService.class, policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MANDATORY)
-    protected void setCDIService(ServiceReference<CDIService> ref) {
-        cdiServiceRef.setReference(ref);
-    }
-
-    protected void unsetCDIService(ServiceReference<CDIService> ref) {
-        cdiServiceRef.unsetReference(ref);
-    }
-
-    private CDIService getCDIService() {
-        return AccessController.doPrivileged(new PrivilegedAction<CDIService>() {
-            @Override
-            public CDIService run() {
-                return cdiServiceRef.getServiceWithException();
-            }
-        });
-
+    @Reference(name = "defaultManagedObjectService", service = DefaultManagedObjectService.class, cardinality = ReferenceCardinality.MANDATORY)
```

Comment:
Forgot to mention this.  Delete the two `@Reference` set methods when using constructor injection for the references.